### PR TITLE
add github cli binary

### PR DESCRIPTION
**Pull request by Moelf**

### CODE REVIEW

File: C/cli/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    GitSource("https://github.com/cli/cli.git", "b2e36a0979a06b94bf364552a856c166cd415234")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd cli/
+make
+mkdir ${bindir}
+mv ./bin/gh ${bindir}/gh
```
Comment (giordano) [inline C/cli/build_tarballs.jl:19]:
```suggestion
mv ./bin/gh ${bindir}/gh${exeext}
```

File: C/cli/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
```
Comment (giordano) [inline C/cli/build_tarballs.jl:24]: Why not FreeBSD?

Comment (Moelf) [reply C/cli/build_tarballs.jl:24]: https://github.com/cli/cli/releases/tag/v1.0.0 they don't build on FreeBSD

Comment (giordano) [reply C/cli/build_tarballs.jl:24]: An error more concrete than "doesn't build"?  All go packages we have here build flawless for all platforms.

Comment (Moelf) [reply C/cli/build_tarballs.jl:24]: What I meant is that 1. they don't have release for FreeBSD in the official repo, and that:
```
[09:05:49] ../../.gopath/pkg/mod/github.com/muesli/termenv@v0.6.0/termenv_unix.go:30:12: undefined: termStatusReport
[09:05:49] ../../.gopath/pkg/mod/github.com/muesli/termenv@v0.6.0/termenv_unix.go:52:12: undefined: termStatusReport
[09:05:49] ../../.gopath/pkg/mod/github.com/muesli/termenv@v0.6.0/termenv_unix.go:76:9: readfds.Bits undefined (type syscall.FdSet has no field or method Bits)
[09:05:49] ../../.gopath/pkg/mod/github.com/muesli/termenv@v0.6.0/termenv_unix.go:87:12: readfds.Bits undefined (type syscall.FdSet has no field or method Bits)
[09:05:52] make: *** [Makefile:30: bin/gh] Error 2
```

Comment (giordano) [reply C/cli/build_tarballs.jl:24]: I guess the patch in https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=248599 might help

Comment (Moelf) [reply C/cli/build_tarballs.jl:24]: how should I try to apply this patch for binary builder? some `sed`?

**Comment by fredrikekre** [comment]
I suggest renaming this to `GitHub_cli` or something. `cli` is just so generic.

**Comment by Moelf** [comment]
gh_cli maybe, since the binary is called `gh`

**Comment by fredrikekre** [comment]
Sure.

**Comment by giordano** [comment]
#1686 to rename to `gh_cli`

**Pull request merged**